config/src/client.rs

```diff
@@ -106,3 +106,166 @@ impl Default for ConfigurationProxy {
         }
     }
 }
+
+impl ConfigurationProxy {
+    /// Finalise Iroha config proxy by instantiating mutually equivalent fields
+    /// via the uppermost Iroha config fields. Configuration fields provided in the
+    /// Iroha config always overwrite those in sumeragi even in case of discrepancy,
+    /// so proper care is advised.
+    ///
+    /// # Errors
+    /// - If the [`self.transaction_time_to_live_ms`] or [`self.transaction_limits.max_wasm_size_bytes`] fields were too small
+    /// - If the [`self.transaction_status_timeout_ms`] field was smaller than [`self.transaction_time_to_live_ms`]
+    /// - If the [`self.torii_api_url`] or [`self.torii_telemetry_url`] were malformed or had the wrong protocol
+    #[allow(clippy::expect_used, clippy::unwrap_in_result)]
+    pub fn finish(&mut self) -> Result<()> {
+        if let Some(tx_ttl) = self.transaction_time_to_live_ms {
+            // Really small TTL would be detrimental to performance
+            if tx_ttl < 500 {
+                eyre::bail!(
+                    ConfigError::ProxyBuildError("Really small `TRANSACTION_TIME_TO_LIVE_MS`, network throughput may be compromised".to_owned())
+                );
+            }
+            // Timeouts bigger than transaction TTL don't make sense as then transaction would be discarded before this timeout
+            if let Some(timeout) = self.transaction_status_timeout_ms {
+                if timeout > tx_ttl {
+                    eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_STATUS_TIMEOUT_MS` bigger than `TRANSACTION_TIME_TO_LIVE_MS`, consider making it smaller".to_owned()));
+                }
+            }
+        }
+        if let Some(tx_limits) = self.transaction_limits {
+            // 1 KiB
+            if tx_limits.max_wasm_size_bytes < 2_u64.pow(10) {
+                eyre::bail!(ConfigError::ProxyBuildError("`TRANSACTION_LIMITS` parameter's `max_wasm_size` field really small, consider making it bigger".to_owned()));
+            }
+        }
+        if let Some(api_url) = &self.torii_api_url {
+            let api_url = api_url.clone().to_string();
+            let split_api_url = api_url.split("://").collect::<Vec<_>>();
```

Comment:
Could be done without allocation, but i wouldn't insist on this. Since we execute this method only once and i remember you suggestion about strong typing.

Reply:
I prefer it this way. The allocation is only a problem when you do several times in a row. Almost everything related to configuration is run once, so flexibility is more important.